frontend/src/metabase/ui/components/overlays/Modal/Modal.stories.mdx

```diff
@@ -0,0 +1,102 @@
+import { useState } from "react";
+import { Canvas, Story, Meta } from "@storybook/addon-docs";
+import { Button, Flex, TextInput } from "metabase/ui";
+import { Modal } from "./";
+
+export const args = {
+  centered: true,
+  fullScreen: false,
+  size: "md",
+};
+
+export const argTypes = {
+  centered: {
+    control: { type: "boolean" },
+  },
+  fullScreen: {
+    control: { type: "boolean" },
+  },
+  title: {
+    control: { type: "text" },
+  },
+  size: {
+    control: {
+      type: "select",
+      options: ["xs", "sm", "md", "lg", "xl", "auto"],
+    },
+  },
+};
+
+<Meta
+  title="Overlays/Modal"
+  component={Modal}
+  args={args}
+  argTypes={argTypes}
+/>
+
+# Modal
+
+Our themed wrapper around [Mantine Modal](https://v6.mantine.dev/core/modal/).
+
+## Docs
+
+- [Mantine Modal Docs](https://v6.mantine.dev/core/modal/)
+
+## Examples
+
+export const SimpleTemplate = args => {
+  const [isOpen, setOpen] = useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+  return (
+    <Flex justify="center">
+      <Button variant="filled" onClick={handleOpen}>
+        Open modal
+      </Button>
+      <Modal {...args} opened={isOpen} onClose={handleClose}>
+        <Text>This is a simple modal with no interactive content</Text>
+      </Modal>
+    </Flex>
+  );
+};
+
+export const FormTemplate = args => {
+  const [isOpen, setOpen] = useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+  return (
+    <Flex justify="center">
+      <Button variant="filled" onClick={handleOpen}>
+        Open modal
+      </Button>
+      <Modal
+        title="New thing"
+        {...args}
+        padding="lg"
+        opened={isOpen}
+        onClose={handleClose}
+      >
+        <form onSubmit={handleClose}>
+          <TextInput label="Name" data-autofocus />
+          <TextInput label="Something else" mt="md" />
+          <Flex direction="row" justify="flex-end" mt="md">
+            <Button onClick={handleClose}>Cancel</Button>
+            <Button type="submit" variant="filled" ml="sm">
+              Create
+            </Button>
+          </Flex>
+        </form>
+      </Modal>
+    </Flex>
+  );
+};
+
+<Canvas>
+  <Story name="Default">{SimpleTemplate}</Story>
+</Canvas>
+
+### Simple
```

Comment:
nit: `Form`?

Reply:
Do you mean `metabase/forms`? It seems it'd also require a `FormProvider` and a full formik setup?

Reply:
I mean that the header label seems wrong. Simple is above this header, and below it is a form example

Reply:
It's just about text on this line